components/authz-service/storage/v2/postgres/postgres_test.go

```diff
@@ -3309,6 +3309,32 @@ func TestCreateRule(t *testing.T) {
 			assert.Nil(t, resp)
 			assert.Error(t, storage_errors.ErrForeignKey, err)
 		},
+		"when rule exists in the applied rules table, return Err": func(t *testing.T) {
+			projID := "project-1"
+			insertTestProject(t, db, projID, "let's go jigglypuff - topsecret", storage.Custom)
+
+			condition1, err := storage.NewCondition(storage.Node, []string{"chef-server-1"}, storage.ChefServer, storage.MemberOf)
+			require.NoError(t, err)
+			rule, err := storage.NewRule("new-id-1", projID, "name", storage.Node, []storage.Condition{condition1})
+			require.NoError(t, err)
+			insertAppliedRule(t, db, rule)
+			resp, err := store.CreateRule(ctx, &rule)
+			assert.Nil(t, resp)
+			assert.Error(t, storage_errors.ErrConflict, err)
+		},
+		"when rule exists in the staging rules table, return Err": func(t *testing.T) {
+			projID := "project-1"
+			insertTestProject(t, db, projID, "let's go jigglypuff - topsecret", storage.Custom)
+
+			condition1, err := storage.NewCondition(storage.Node, []string{"chef-server-1"}, storage.ChefServer, storage.MemberOf)
+			require.NoError(t, err)
+			rule, err := storage.NewRule("new-id-1", projID, "name", storage.Node, []storage.Condition{condition1})
+			insertStagedRule(t, db, rule)
+			require.NoError(t, err)
+			resp, err := store.CreateRule(ctx, &rule)
+			assert.Nil(t, resp)
+			assert.Error(t, storage_errors.ErrConflict, err)
```

Comment:
⚠️ I don't think that's how it works: [`assert.Error`](https://godoc.org/github.com/stretchr/testify/assert#Error) has type `func Error(t TestingT, err error, msgAndArgs ...interface{}) bool`, so, what this line does is:

**it's checking that `storage_errors.ErrConflict` is indeed an error, and it's using `err` as extra information to display for when it isn't**

I think we want either
```go
assert.Error(t, err)
```
or
```go
assert.Equal(t, storage_errors.ErrConflict, err)
```
here.